### PR TITLE
Fk14 - UI changes in show page

### PR DESCRIPTION
**Pull request by jaspreet21anand**

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_requirement, only: :create
+
+  def create
+    @comment = @requirement.comments.build(comment_params.merge(user_id: current_user.id))
+    redirect_to @requirement, alert: 'Could not add comment' unless @comment.save
+  end
+
+  def destroy
```
Comment (ShubhamGupta) [inline app/controllers/comments_controller.rb]: Remove the unused action.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_requirement, only: :create
+
+  def create
+    @comment = @requirement.comments.build(comment_params.merge(user_id: current_user.id))
+    redirect_to @requirement, alert: 'Could not add comment' unless @comment.save
```
Comment (ShubhamGupta) [inline app/controllers/comments_controller.rb]: Display both success as well as failure messages. Always.

File: app/controllers/donor_requirements_controller.rb [hunk 3]
```diff
@@ -1,28 +1,27 @@
 class DonorRequirementsController < ApplicationController
-  include Donation
 
-  before_action :set_requirement
+  before_action :set_requirement, only: [:create, :destroy]
+  before_action :dont_allow_owner, only: :create
```
Comment (ShubhamGupta) [inline app/controllers/donor_requirements_controller.rb]: rename to `restrict_owner`

File: app/controllers/donor_requirements_controller.rb [hunk 4]
```diff
@@ -1,28 +1,27 @@
...
   before_action :authenticate_user!
 
   def create
     @donor_requirement = DonorRequirement.new(requirement_id: params[:requirement_id], donor_id: current_user.id)
```
Comment (ShubhamGupta) [inline app/controllers/donor_requirements_controller.rb]: Use `@requirement.donor_requirements.build...`

File: app/controllers/donor_requirements_controller.rb [hunk 5]
```diff
@@ -1,28 +1,27 @@
...
-      flash.now[:alert] = "Couldn't show interest"
+      flash[:alert] = "Couldn't show interest"
     end
-    load_donations
-    render 'requirements/toggle_interest'
+    redirect_to @requirement
   end
 
   def destroy
     @donor_requirement = DonorRequirement.find_by(requirement_id: params[:requirement_id], donor_id: current_user.id)
```
Comment (ShubhamGupta) [inline app/controllers/donor_requirements_controller.rb]: Same as above.

File: config/routes.rb [hunk 6]
```diff
@@ -19,6 +19,7 @@
       put 'fulfill'
       put 'reject_donor', as: :reject_current_donor_of
     end
+    resources :comments, only: [:create, :destroy]
```
Comment (ShubhamGupta) [inline config/routes.rb]: Remove `destroy`

File: app/helpers/application_helper.rb [hunk 7]
```diff
@@ -1,2 +1,7 @@
 module ApplicationHelper
+
+  def owner
+    @requirement.requestor_id == current_user.id
```
Comment (ShubhamGupta) [inline app/helpers/application_helper.rb]: Incorrect method. How can you make sure @requirement exists?

File: app/models/requirement.rb [hunk 8]
```diff
@@ -91,7 +92,9 @@ def update_donor_and_reject_interested_donors
 
   def attach_display_image(image_file)
     image = create_image(attachment: image_file)
-    image.update_column(:attacheable_type, :Image)
+    if image.id
```
Comment (ShubhamGupta) [inline app/models/requirement.rb:95]: Whats the use of this?

File: app/helpers/application_helper.rb (hunk 7 above)
Comment (jaspreet21anand) [inline app/helpers/application_helper.rb]: This method is in helper class and used in views, so control will reach only after @requirement exists.

File: app/models/requirement.rb (hunk 8 above)
Comment (jaspreet21anand) [inline app/models/requirement.rb:95]: if the validation of image fails, it is not saved. Hence this check prevents from raising exception on calling update_column on a object which is not in db

**Pull request merged**